src/deploy/functions/prepare.ts

```diff
@@ -56,7 +57,7 @@ export async function prepare(
   const wantBackend = await runtimeDelegate.discoverSpec(runtimeConfig, env);
   payload.functions = { backend: wantBackend };
   if (backend.isEmptyBackend(wantBackend)) {
-    return;
+    throw new FirebaseError("Found nothing to deploy. Is your functions source empty?");
```

Comment:
We traditionally have explicitly not errored on an empty backend. This allows a customer to say `firebase init`, initialize the kitchen sink, and then still deploy before they get to editing their functions.

Reply:
This is was a band-aid patch on current bug - right now, the Firebase CLI throws and error on empty `index.js` file:

```
$ firebase deploy --only functions
=== Deploying to '...'..

i  deploying functions
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
✔  functions: required API cloudfunctions.googleapis.com is enabled
✔  functions: required API cloudbuild.googleapis.com is enabled

Error: An unexpected error has occurred.

Having trouble? Try again or contact support with contents of firebase-debug.log
```

It sounds like this is a bad fix. I'll revert this change and make a separate PR to fix the issue for empty backends.